### PR TITLE
Match .luau filename extension by default

### LINKED ISSUE #325: Support `.luau` as an extension for files when Roblox feature is active
Luau has formally adopted `.luau` as it's extension (see: https://github.com/Roblox/luau/issues/97) so we should support it in Selene too.

### PR DESCRIPTION
**Pull request by Yumacide**
Closes #325.
It seems like the glob crate doesn't allow pattern lists like `**/*.{lua,luau}` (see [docs](https://docs.rs/glob/latest/glob/struct.Pattern.html)), so I chose to match multiple patterns instead. ~~I also renamed `--pattern` to `--patterns` to match the change, let me know if that's not okay.~~

### CODE REVIEW

File: docs/src/cli/usage.md [hunk 1]
```diff
@@ -72,10 +73,10 @@ Results:
 0 parse errors
 ```
 
-**--num-threads** *num-threads*
+**--num-threads** _num-threads_
```
Comment (Kampfkarren) [inline docs/src/cli/usage.md:76]: Revert the unrelated markdown changes (this and some whitespace stuff)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,13 @@
 # Changelog
 This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased]
+
+### Changed
+
+- Match `.luau` filename extension by default.
+- Allow multiple comma-separated glob patterns with `--patterns` rather than `--pattern`.
```
Comment (Kampfkarren) [inline CHANGELOG.md:9]: `--pattern` should just take multiple patterns, I don't want to break backwards compatibility with the older parameter, and commas already aren't allowed in filenames in any of the operating systems we support, from what I know

File: docs/src/cli/usage.md [hunk 3]
```diff
@@ -72,10 +73,10 @@ Results:
...
 
 Specifies the number of threads for selene to use. Defaults to however many cores your CPU has. If you type `selene --help`, you can see this number because it will show as the default for you.
 
-**--pattern** *pattern*
+**--patterns** _patterns_
 
-A [glob](https://en.wikipedia.org/wiki/Glob_(programming)) to match what files selene should check for. For example, if you only wanted to check files that end with `.spec.lua`, you would input `--pattern **/*.spec.lua`. Defaults to `**/*.lua`, meaning "any lua file".
+Comma-separated [globs](<https://en.wikipedia.org/wiki/Glob_(programming)>) to match what files selene should check for. For example, if you only wanted to check files that end with `.spec.lua`, you would input `--patterns **/*.spec.lua`. Defaults to `**/*.lua,**/*.luau`, meaning "any lua/luau file".
```
Comment (Kampfkarren) [inline docs/src/cli/usage.md:82]: Why the `<>`?

File: selene/src/opts.rs [hunk 4]
```diff
@@ -7,9 +7,9 @@ use structopt::{clap::arg_enum, StructOpt};
 #[structopt(setting(structopt::clap::AppSettings::ArgsNegateSubcommands))]
 #[structopt(setting(structopt::clap::AppSettings::SubcommandsNegateReqs))]
 pub struct Options {
-    /// A glob to match files with to check
-    #[structopt(long, default_value = "**/*.lua")]
-    pub pattern: String,
+    /// Comma-separated globs to match files to check
+    #[structopt(long, default_value = "**/*.lua,**/*.luau")]
```
Comment (Kampfkarren) [inline selene/src/opts.rs:11]: We probably shouldn't default match `.luau` unless the "roblox" feature flag is enabled (which will eventually be renamed to Luau).

You can do this with something like:

```suggestion
    #[cfg_attr(feature = "roblox", structopt(long, default_value = "**/*.lua,**/*.luau"))]
    #[cfg_attr(not(feature = "roblox"), structopt(long, default_value = "**/*.lua,**/*.luau"))]
```

**Review by Kampfkarren — CHANGES_REQUESTED** [review]
Neat!

File: docs/src/cli/usage.md (hunk 3 above)
Comment (Yumacide) [reply docs/src/cli/usage.md:82]: This and the whitespace changes were done automatically by my editor

File: CHANGELOG.md (hunk 2 above)
Comment (Yumacide) [reply CHANGELOG.md:9]: How would I do that?

Comment (Kampfkarren) [reply CHANGELOG.md:9]: Just don't change the name :P

Comment (Yumacide) [reply CHANGELOG.md:9]: Oh, I thought you were suggesting something like `selene --pattern foo --pattern foo`.

**Comment by Yumacide** [comment]
Ready to merge! In retrospect, this would be best as an upstream change to `glob`, but I can't find any discussion on it there.

**Comment by Kampfkarren** [comment]
I think glob already supports this with some weird syntax with pipelines, it's not as intuitive

**Comment by Yumacide** [comment]
If that's the case, it's not documented: https://docs.rs/glob/latest/glob/struct.Pattern.html

**Comment by Yumacide** [comment]
Turns out commas are allowed in filenames on windows & *nix -- maybe it's a better idea to do multiple pattern args like `selene --pattern "**/*.lua" --pattern "**/*.luau"` after all.

**Comment by Kampfkarren** [comment]
Yeah that'll work

File: selene/src/opts.rs [hunk 5]
```diff
@@ -1,15 +1,34 @@
...
+use std::{ffi::OsString, path::PathBuf, str::FromStr};
 
 use structopt::{clap::arg_enum, StructOpt};
 
+#[derive(Clone, Debug)]
+pub struct Patterns {
+    pub vec: Vec<String>,
+}
+
+impl FromStr for Patterns {
```
Comment (Yumacide) [inline selene/src/opts.rs:10]: I had to do this because structopt's `default_value` has to be a string literal; I can't just make it a `Vec<String>` like it should normally be.

File: selene/src/opts.rs [hunk 6]
```diff
@@ -1,15 +1,34 @@
...
 #[structopt(rename_all = "kebab-case")]
 #[structopt(setting(structopt::clap::AppSettings::ArgsNegateSubcommands))]
 #[structopt(setting(structopt::clap::AppSettings::SubcommandsNegateReqs))]
 pub struct Options {
     /// A glob to match files with to check
-    #[structopt(long, default_value = "**/*.lua")]
-    pub pattern: String,
+    #[cfg_attr(
+        feature = "roblox",
+        structopt(long, default_value = "**/*.lua\0**/*.luau")
```
Comment (Yumacide) [inline selene/src/opts.rs:28]: It seems like `\0` and `/` are the only universally prohibited characters in filenames

Comment (Yumacide) [reply selene/src/opts.rs:28]: This is just for the default value, though. Users would just pass `--pattern` multiple times

File: selene/src/main.rs [hunk 7]
```diff
@@ -347,9 +347,15 @@ fn read_file(checker: &Checker<toml::value::Value>, filename: &Path) {
     );
 }
 
-fn start(matches: opts::Options) {
+fn start(mut matches: opts::Options) {
     *OPTIONS.write().unwrap() = Some(matches.clone());
 
+    if matches.pattern.is_empty() {
```
Comment (Kampfkarren) [inline selene/src/main.rs:353]: FWIW this is what I was going to suggest 😄
